pkg/resmap/resmap_test.go

```diff
@@ -70,6 +70,202 @@ metadata:
+				},
+			}),
+		resid.NewResIdWithPrefixNamespace(cmap, "cm2", "prefix1", "ns1"): resource.NewResourceFromMap(
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm2",
+				},
+			}),
+	}
+
+	_, ok := rm1.DemandOneMatchForId(resid.NewResIdWithPrefixNamespace(cmap, "cm2", "prefix1", "ns1"))
+	if !ok {
+		t.Fatal("Expected single map entry but got none")
+	}
+
+	// confirm that ns and prefix are not included in match
+	_, ok = rm1.DemandOneMatchForId(resid.NewResIdWithPrefixNamespace(cmap, "cm2", "prefix", "ns"))
+	if !ok {
+		t.Fatal("Expected single map entry but got none")
+	}
+
+	// confirm that name is matched correctly
+	result, ok := rm1.DemandOneMatchForId(resid.NewResIdWithPrefixNamespace(cmap, "cm3", "prefix1", "ns1"))
+	if ok {
+		t.Fatalf("Expected no map entries but got %v", result)
+	}
+
+	cmap2 := gvk.Gvk{Version: "v2", Kind: "ConfigMap"}
+
+	// confirm that gvk is matched correctly
+	result, ok = rm1.DemandOneMatchForId(resid.NewResIdWithPrefixNamespace(cmap2, "cm2", "prefix1", "ns1"))
+	if ok {
+		t.Fatalf("Expected no map entries but got %v", result)
+	}
+
+}
+
+func TestFilterBy(t *testing.T) {
+	rm := ResMap{resid.NewResIdWithPrefixNamespace(cmap, "cm1", "prefix1", "ns1"): resource.NewResourceFromMap(
+		map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "ConfigMap",
+			"metadata": map[string]interface{}{
+				"name": "cm1",
+			},
+		}),
+		resid.NewResIdWithPrefixNamespace(cmap, "cm2", "prefix1", "ns1"): resource.NewResourceFromMap(
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm2",
+				},
+			}),
+	}
+	rm1 := ResMap{
+		resid.NewResIdWithPrefixNamespace(cmap, "cm3", "prefix1", "ns2"): resource.NewResourceFromMap(
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm2",
+				},
+			}),
+	}
+
+	for k, v := range rm {
+		rm1[k] = v
+	}
+
+	empty := rm1.FilterBy(resid.NewResIdWithPrefixNamespace(cmap, "cm4", "prefix1", "ns3"))
+	if len(empty) != 0 {
+		t.Fatalf("Expected empty filtered map but got %v", empty)
+	}
+
+	ns1map := rm1.FilterBy(resid.NewResIdWithPrefixNamespace(cmap, "cm4", "prefix1", "ns1"))
+	if !reflect.DeepEqual(rm, ns1map) {
+		t.Fatalf("Expected %v but got back %v", rm, ns1map)
+	}
+}
+func TestDeepCopy(t *testing.T) {
+	rm1 := ResMap{
+		resid.NewResId(cmap, "cm1"): resource.NewResourceFromMap(
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm1",
+				},
+			}),
+		resid.NewResId(cmap, "cm2"): resource.NewResourceFromMap(
+			map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "cm2",
+				},
+			}),
+	}
+
+	rm2 := rm1.DeepCopy()
```

Comment:
Can you also make sure the resources in `rm1` and `rm2` are different objects?

Reply:
done